pkg/coveragedb/spanner.go

```diff
@@ -89,6 +90,64 @@ func SaveMergeResult(ctx context.Context, projectID string, covMap map[string]*C
 	return nil
 }
 
+type linesCoverage struct {
+	LinesInstrumented []int64
+	HitCounts         []int64
+}
+
+func linesCoverageStmt(ns, filepath, commit string, timePeriod TimePeriod) spanner.Statement {
+	return spanner.Statement{
+		SQL: `
+select
+	linesinstrumented,
+	hitcounts
+from merge_history
+	join files
+		on merge_history.session = files.session
+where
+	namespace=$1 and dateto=$2 and duration=$3 and filepath=$4 and commit=$5`,
+		Params: map[string]interface{}{
+			"p1": ns,
+			"p2": timePeriod.DateTo,
+			"p3": timePeriod.Days,
+			"p4": filepath,
+			"p5": commit,
+		},
+	}
+}
+
+func ReadLinesHitCount(ctx context.Context, ns, commit, file string, tp TimePeriod,
+) (map[int]int, error) {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	client, err := NewClient(ctx, projectID)
```

Comment:
Probably not super important at this stage, but eventually it would be nice to keep just one spanner connection per GAE instance -- we don't need to create one per each request/each function.

Reply:
Let me resolve it to unblock merging.